src/brasil/gov/portal/tests/robot/test_upload.robot

```diff
@@ -0,0 +1,70 @@
+*** Settings ***
+
+Resource  plone/app/robotframework/keywords.robot
+Variables  plone/app/testing/interfaces.py
+Library  Remote  ${PLONE_URL}/RobotRemote
+
+Test Setup  Open test browser
+Test Teardown  Close all browsers
+
+*** Variables ***
+
+@{upload_overlay_selector}  css=form.fileupload
+@{valid_types} =  image.jpg
+@{invalid_types} =  file.mp3  file.ogg
+
+*** Test cases ***
+
+Test Upload Valid Types
+    [Documentation]  Uploading image files is allowed by default
+
+    Enable Autologin as  Site Administrator
+    Goto Homepage
+
+    Add Files  @{valid_types}
+    Start Upload
+
+    # we just assume the file was successfully upload
+
+Test Upload Invalid Types
+    [Documentation]  Uploading audio files is not allowed by default
+
+    Enable Autologin as  Site Administrator
+    Goto Homepage
+
+    Add Files  @{invalid_types}
+
+    # FIXME: Firefox text enconding seems to be wrong
+    #        and displays "nÃ£o" instead of "não"
+    #        we need to check on the ff_profile directory
+    #        using this workaround meanwhile
```

Comment:
It looks like a Javascript bug, it is not decoding utf-8 strings.
Upload translations are declared at [Media Upload Viewlet](https://github.com/collective/collective.upload/blob/master/src/collective/upload/browser/templates/media_uploader.pt#L18-L39), then Javascript insert the translations into [Fileupload](https://github.com/collective/collective.upload/blob/master/webpack/app/upload.js#L73-L80).

My bet is that we need to [decode utf8](https://stackoverflow.com/a/13691499) in this Javascript code.

And this should be fixed int collective.upload package.